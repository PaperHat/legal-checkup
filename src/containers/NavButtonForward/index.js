import { connect } from 'react-redux'

import WithOnClickIfUnlocked from '../../components/WithOnClickIfUnlocked'
import StyledNavButtonForward from '../../components/styled/NavButton/Forward'
import { getActiveQuestion, getCount, nextQuestion } from '../../state'

function mapStateToProps (state) {
  return {
    unlocked: getActiveQuestion(state) < getCount(state) + 1
  }
}

function mapDispatchToProps (dispatch) {
  return {
    handleClick: () => dispatch(nextQuestion())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(
  WithOnClickIfUnlocked(StyledNavButtonForward)
)
