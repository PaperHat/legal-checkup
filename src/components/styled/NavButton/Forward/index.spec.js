import React from 'react'
import { shallow } from 'enzyme'

import NavButtonForward from '.'

describe('components:styled:NavButton:NavButtonForward', () => {
  it('renders the Back and CSS properties properly', () => {
    expect(
      toJson(shallow(<NavButtonForward>&#x25B6;</NavButtonForward>))
    ).toMatchSnapshot()
  })
})
