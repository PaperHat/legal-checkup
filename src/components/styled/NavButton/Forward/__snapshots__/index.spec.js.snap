// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:NavButton:NavButtonForward renders the Back and CSS properties properly 1`] = `
.c0 {
  background-color: #ededed;
  border: none;
  color: #c0c0c0;
  height: 42px;
  outline: none;
  width: 48px;
  border-radius: 0 5px 5px 0;
}

<button
  className="c0"
>
  â–¶
</button>
`;
